Xamarin.Forms.Core/Setter.cs

```diff
@@ -47,59 +60,76 @@ object IValueProvider.ProvideValue(IServiceProvider serviceProvider)
 		{
-			if (target == null)
-				throw new ArgumentNullException(nameof(target));
+			var targetObject = FindTargetObject(scope, Target);
+			if (targetObject == null)
+				throw new ArgumentNullException(nameof(targetObject));
 			if (Property == null)
 				return;
 
-			object originalValue = target.GetValue(Property);
+			object originalValue = targetObject.GetValue(Property);
 			if (!Equals(originalValue, Property.DefaultValue))
 			{
-				_originalValues.Remove(target);
-				_originalValues.Add(target, originalValue);
+				_originalValues.Remove(targetObject);
+				_originalValues.Add(targetObject, originalValue);
 			}
 
 			var dynamicResource = Value as DynamicResource;
-			var binding = Value as BindingBase;
-			if (binding != null)
-				target.SetBinding(Property, binding.Clone(), fromStyle);
+			if (Value is BindingBase binding)
+				targetObject.SetBinding(Property, binding.Clone(), fromStyle);
 			else if (dynamicResource != null)
-				target.SetDynamicResource(Property, dynamicResource.Key, fromStyle);
+				targetObject.SetDynamicResource(Property, dynamicResource.Key, fromStyle);
 			else
 			{
 				if (Value is IList<VisualStateGroup> visualStateGroupCollection)
-					target.SetValue(Property, visualStateGroupCollection.Clone(), fromStyle);
+					targetObject.SetValue(Property, visualStateGroupCollection.Clone(), fromStyle);
 				else
-					target.SetValue(Property, Value, fromStyle);
+					targetObject.SetValue(Property, Value, fromStyle);
 			}
 		}
 
-		internal void UnApply(BindableObject target, bool fromStyle = false)
+		internal void UnApply(BindableObject scope, bool fromStyle = false)
 		{
-			if (target == null)
-				throw new ArgumentNullException(nameof(target));
+			var targetObject = FindTargetObject(scope, Target);
+			if (targetObject == null)
+				throw new ArgumentNullException(nameof(targetObject));
 			if (Property == null)
 				return;
 
-			object actual = target.GetValue(Property);
+			object actual = targetObject.GetValue(Property);
 			if (!Equals(actual, Value) && !(Value is Binding) && !(Value is DynamicResource))
 			{
 				//Do not reset default value if the value has been changed
-				_originalValues.Remove(target);
+				_originalValues.Remove(targetObject);
 				return;
 			}
 
-			object defaultValue;
-			if (_originalValues.TryGetValue(target, out defaultValue))
+			if (_originalValues.TryGetValue(targetObject, out object defaultValue))
 			{
 				//reset default value, unapply bindings and dynamicResource
-				target.SetValue(Property, defaultValue, fromStyle);
-				_originalValues.Remove(target);
+				targetObject.SetValue(Property, defaultValue, fromStyle);
+				_originalValues.Remove(targetObject);
 			}
 			else
-				target.ClearValue(Property);
+				targetObject.ClearValue(Property);
+		}
+
+		BindableObject FindTargetObject(BindableObject scope, object target)
+		{
+			if (scope == null || target == null)
+				return scope;
```

Comment:
Is this necessary? I think the next 2 if statements will catch any nulls. The only case that looks to change is if scope is null, but target is BindableObject. But, we are returning the object directly.